report/util_test.go

```diff
@@ -316,7 +316,168 @@ func TestDiff(t *testing.T) {
 	}
 
 	for i, tt := range tests {
-		diff, _ := diff(tt.inCurrent, tt.inPrevious)
+		diff, _ := plusDiff(tt.inCurrent, tt.inPrevious)
+		for _, actual := range diff {
+			if !reflect.DeepEqual(actual.ScannedCves, tt.out.ScannedCves) {
+				h := pp.Sprint(actual.ScannedCves)
+				x := pp.Sprint(tt.out.ScannedCves)
+				t.Errorf("[%d] cves actual: \n %s \n expected: \n %s", i, h, x)
+			}
+
+			for j := range tt.out.Packages {
+				if !reflect.DeepEqual(tt.out.Packages[j], actual.Packages[j]) {
+					h := pp.Sprint(tt.out.Packages[j])
+					x := pp.Sprint(actual.Packages[j])
+					t.Errorf("[%d] packages actual: \n %s \n expected: \n %s", i, x, h)
+				}
+			}
+		}
+	}
+}
+
+func TestMinusDiff(t *testing.T) {
+	atCurrent, _ := time.Parse("2006-01-02", "2014-12-31")
+	atPrevious, _ := time.Parse("2006-01-02", "2014-11-31")
+	var tests = []struct {
+		inCurrent  models.ScanResults
+		inPrevious models.ScanResults
+		out        models.ScanResult
+	}{
+		{
+			inCurrent: models.ScanResults{
+				{
+					ScannedAt:  atCurrent,
+					ServerName: "u16",
+					Family:     "ubuntu",
+					Release:    "16.04",
+					ScannedCves: models.VulnInfos{
+						"CVE-2012-6702": {
+							CveID:            "CVE-2012-6702",
+							AffectedPackages: models.PackageFixStatuses{{Name: "libexpat1"}},
+							DistroAdvisories: []models.DistroAdvisory{},
+							CpeURIs:          []string{},
+						},
+						"CVE-2014-9761": {
+							CveID:            "CVE-2014-9761",
+							AffectedPackages: models.PackageFixStatuses{{Name: "libc-bin"}},
+							DistroAdvisories: []models.DistroAdvisory{},
+							CpeURIs:          []string{},
+						},
+					},
+					Packages: models.Packages{},
+					Errors:   []string{},
+					Optional: map[string]interface{}{},
+				},
+			},
+			inPrevious: models.ScanResults{
+				{
+					ScannedAt:  atPrevious,
+					ServerName: "u16",
+					Family:     "ubuntu",
+					Release:    "16.04",
+					ScannedCves: models.VulnInfos{
+						"CVE-2012-6702": {
+							CveID:            "CVE-2012-6702",
+							AffectedPackages: models.PackageFixStatuses{{Name: "libexpat1"}},
+							DistroAdvisories: []models.DistroAdvisory{},
+							CpeURIs:          []string{},
+						},
+						"CVE-2014-9761": {
+							CveID:            "CVE-2014-9761",
+							AffectedPackages: models.PackageFixStatuses{{Name: "libc-bin"}},
+							DistroAdvisories: []models.DistroAdvisory{},
+							CpeURIs:          []string{},
+						},
+					},
+					Packages: models.Packages{},
+					Errors:   []string{},
+					Optional: map[string]interface{}{},
+				},
+			},
+			out: models.ScanResult{
+				ScannedAt:   atCurrent,
+				ServerName:  "u16",
+				Family:      "ubuntu",
+				Release:     "16.04",
+				Packages:    models.Packages{},
```

Comment:
Please remove the initialization of the empty structure as it is easier to understand the intent of the test case (noise removal).